controllers/infrastructure/byomachine_controller_test.go

```diff
@@ -379,6 +408,71 @@ var _ = Describe("Controllers/ByomachineController", func() {
 						Expect(err).To(MatchError(fmt.Sprintf("byomachines.infrastructure.cluster.x-k8s.io %q not found", byoMachineLookupKey.Name)))
 					})
 				})
+
+				Context("When installer config exists", func() {
+					BeforeEach(func() {
+						k8sInstallerConfig = builder.K8sInstallerConfig(defaultNamespace, "").
+							WithName(byoMachine.Name).
+							WithBundleRepo("projects.registry.vmware.com/cluster_api_provider_bringyourownhost").
+							WithBundleType("k8s").
+							Build()
+						Expect(k8sClientUncached.Create(ctx, k8sInstallerConfig)).Should(Succeed())
+						WaitForObjectsToBePopulatedInCache(k8sInstallerConfig)
+
+						ph, err := patch.NewHelper(byoMachine, k8sClientUncached)
+						Expect(err).ShouldNot(HaveOccurred())
+						byoMachine.Spec.InstallerRef = &corev1.ObjectReference{
+							Kind:       "K8sInstallerConfigTemplate",
+							Namespace:  defaultNamespace,
+							Name:       defaultK8sInstallerConfigTemplateName,
+							APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+						}
+						Expect(ph.Patch(ctx, byoMachine, patch.WithStatusObservedGeneration{})).Should(Succeed())
+
+						WaitForObjectToBeUpdatedInCache(byoMachine, func(object client.Object) bool {
+							return object.(*infrastructurev1beta1.ByoMachine).Spec.InstallerRef != nil
+						})
+					})
+
+					It("should fail if installer config is not ready", func() {
```

Comment:
It should requeue in installer config is not ready

Reply:
done